app/main.py

```diff
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for person_dict in people:
+        # Create a Person instance and add it to the list
+        person = Person(person_dict["name"], person_dict["age"])
+        person_list.append(person)
+
+    for person_dict in people:
+        name = person_dict["name"]
+        if "wife" in person_dict and person_dict["wife"]:
+            person = Person.people[name]
+            spouse = Person.people[person_dict["wife"]]
```

Comment:
There is a potential issue here if the spouse's name is not found in the `Person.people` dictionary. This will raise a `KeyError`. Consider adding a check to ensure the spouse exists before attempting to access it.